File: backend/src/main/java/top/vuhe/common/exception/SystemProcessingExceptionHandler.java
package top.vuhe.common.exception;

import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import top.vuhe.common.ApiResponse;

/**
 * 统一异常处理器
 *
 * @author zhuhe
 */
@Slf4j
@RestControllerAdvice
public class SystemProcessingExceptionHandler {
    /**
     * 处理自定义系统异常
     *
     * @param e 异常
     * @return 错误信息
     */
    @ExceptionHandler(SystemProcessingException.class)
    public ApiResponse<?> handleMyException(SystemProcessingException e) {
        return ApiResponse.ofException(e);
    }

    /**
     * 拦截其它错误
     *
     * @param e 异常
     * @return 错误信息
     */
    @ExceptionHandler(Exception.class)
    public ApiResponse<?> handleException(Exception e) {
        log.error(e.getMessage(), e);
        return ApiResponse.ofErrorEnum(ExceptionEnum.UNKNOWN);
    }
}

File: backend-kt/src/main/kotlin/top/vuhe/sw/common/exception/SystemProcessingExceptionHandler.kt
package top.vuhe.sw.common.exception

import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice
import top.vuhe.sw.common.ApiResponse
import top.vuhe.sw.common.util.Slf4j
import top.vuhe.sw.common.util.Slf4j.Companion.log

/**
 * 统一异常处理器
 *
 * @author vuhe
 */
@Slf4j
@RestControllerAdvice
class SystemProcessingExceptionHandler {
    /**
     * 处理自己抛出的异常
     */
    @ExceptionHandler(SystemProcessingException::class)
    fun handleMyException(e: SystemProcessingException): ApiResponse<*> {
        return ApiResponse.ofException(e)
    }

    /**
     * 其它异常拦截
     */
    @ExceptionHandler(Exception::class)
    fun handleException(e: Exception): ApiResponse<*> {
        log.error(e.message, e)
        return ApiResponse.ofErrorEnum(ExceptionEnum.UNKNOWN)
    }
}